presto-spark-base/src/test/java/com/facebook/presto/spark/task/TestPrestoSparkTaskExecution.java

```diff
@@ -0,0 +1,133 @@
+
+import com.facebook.presto.Session;
+import com.facebook.presto.execution.ScheduledSplit;
+import com.facebook.presto.execution.TaskId;
+import com.facebook.presto.execution.TaskSource;
+import com.facebook.presto.execution.TaskStateMachine;
+import com.facebook.presto.execution.TaskTestUtils;
+import com.facebook.presto.execution.executor.TaskExecutor;
+import com.facebook.presto.execution.scheduler.TableWriteInfo;
+import com.facebook.presto.metadata.Split;
+import com.facebook.presto.operator.TaskContext;
+import com.facebook.presto.spark.execution.PrestoSparkTaskExecution;
+import com.facebook.presto.spi.ConnectorId;
+import com.facebook.presto.sql.planner.LocalExecutionPlanner;
+import com.facebook.presto.sql.planner.PlanFragment;
+import com.facebook.presto.sql.planner.TestingOutputBuffer;
+import com.facebook.presto.sql.planner.TestingRemoteSourceFactory;
+import com.facebook.presto.testing.TestingSplit;
+import com.facebook.presto.testing.TestingTaskContext;
+import com.facebook.presto.testing.TestingTransactionHandle;
+import com.google.common.base.Ticker;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.units.DataSize;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static com.facebook.airlift.concurrent.Threads.daemonThreadsNamed;
+import static com.facebook.presto.SystemSessionProperties.NATIVE_EXECUTION_ENABLED;
+import static com.facebook.presto.execution.TaskManagerConfig.TaskPriorityTracking.TASK_FAIR;
+import static com.facebook.presto.execution.TaskTestUtils.TABLE_SCAN_NODE_ID;
+import static com.facebook.presto.execution.TaskTestUtils.createPlanFragment;
+import static com.facebook.presto.execution.TaskTestUtils.createTestingPlanner;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.testing.assertions.Assert.assertEquals;
+import static com.facebook.presto.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+import static io.airlift.units.DataSize.Unit.GIGABYTE;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static java.util.concurrent.Executors.newScheduledThreadPool;
+
+public class TestPrestoSparkTaskExecution
+{
+    Session nativeTestSession;
+    Session nonNativeTestSession;
+    LocalExecutionPlanner.LocalExecutionPlan localExecutionPlan;
+    TaskExecutor taskExecutor;
+    PlanFragment planFragment = createPlanFragment();
+    ExecutorService taskNotificationExecutor;
+    ScheduledExecutorService scheduledExecutor;
+    TaskStateMachine taskStateMachine;
+    ScheduledSplit split1 = new ScheduledSplit(1, TABLE_SCAN_NODE_ID, new Split(new ConnectorId("test"), TestingTransactionHandle.create(), TestingSplit.createLocalSplit()));
```

Comment:
Do we need all these split1, split2,... variables? Can we create a single List<ScheduledSplit> splits = ImmutableList.of(....) and use in both tests? Any particular reason Java test uses splits 1, 2, 3 and Native uses splits 4, 5?

Reply:
Actually, they don't require different splits anymore. 
I was initially having trouble having both of them in same test as I was not able to trigger a proper tearDown after each method. This is fixed now.